autoendpoint/src/routers/fcm/router.rs

```diff
@@ -178,46 +176,40 @@ impl Router for FcmRouter {
         // transaction. According to the documentation, GCM and FCM are interoperable,
         // meaning that the client provided tokens should match up, and that the structure
-        // of the data should also not make a huge difference. (Although, that's untested.)
+        // of the data should also not make a huge difference.
         //
-        match RouterType::from_str(&notification.subscription.user.router_type)
-            .unwrap_or(RouterType::FCM)
-        {
-            RouterType::GCM => {
-                trace!("Sending message to GCM: [{:?}]", &app_id);
-                if let Err(e) = client.send_gcm(message_data, routing_token, ttl).await {
-                    return Err(handle_error(
-                        e,
-                        &self.metrics,
-                        self.db.as_ref(),
-                        "gcm",
-                        &app_id,
-                        notification.subscription.user.uaid,
-                    )
-                    .await);
-                }
-                incr_success_metrics(&self.metrics, "gcm", &app_id, notification);
-            }
-            _ => {
-                trace!("Sending message to FCM: [{:?}]", &app_id);
-                if let Err(e) = client.send(message_data, routing_token, ttl).await {
-                    return Err(handle_error(
-                        e,
-                        &self.metrics,
-                        self.db.as_ref(),
-                        "fcmv1",
-                        &app_id,
-                        notification.subscription.user.uaid,
-                    )
-                    .await);
-                }
-                incr_success_metrics(&self.metrics, "fcmv1", &app_id, notification);
-            }
-        }
+        // In Oct of 2023, "legacy" GCM support was deprecated. After some
+        // research, it was hypothesised that the current FCM project may be able
+        // to accept GCM routing identifiers as FCM message tokens. The current code presumes
+        // that this is a fragile relationship and so GCM errors do not immediately drop the
+        // remote user's endpoint (potentially causing a UA reset.)
+        //
+        trace!(
+            "Sending message to {platform}: [{:?}]",
+            &app_id,
+            platform = if client.is_gcm { "GCM as FCM" } else { "FCM" }
+        );
+        if let Err(e) = client.send(message_data, routing_token, ttl).await {
+            return Err(handle_error(
+                e,
+                &self.metrics,
+                self.db.as_ref(),
+                if client.is_gcm { "gcm_as_fcm" } else { "fcm" },
+                &app_id,
+                notification.subscription.user.uaid,
+            )
+            .await);
+        };
+        incr_success_metrics(
+            &self.metrics,
+            if client.is_gcm { "gcm_as_fcm" } else { "fcm" },
+            &app_id,
+            notification,
+        );
```

Comment:
I think we should stick with the current "fcmv1" for now, especially being this is initially canary only.

```suggestion
        let platform = if client.is_gcm { "gcm_as_fcmv1" } else { "fcmv1" };
        trace!(
            "Sending message to {platform}: [{:?}]",
            &app_id
        );
        if let Err(e) = client.send(message_data, routing_token, ttl).await {
            return Err(handle_error(
                e,
                &self.metrics,
                self.db.as_ref(),
                platform,
                &app_id,
                notification.subscription.user.uaid,
            )
            .await);
        };
        incr_success_metrics(
            &self.metrics,
            platform,
            &app_id,
            notification,
        );
```